test/unit/index/test-rangesegment.cpp

```diff
@@ -0,0 +1,84 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC.
+//
+// Produced at the Lawrence Livermore National Laboratory
+//
+// LLNL-CODE-689114
+//
+// All rights reserved.
+//
+// This file is part of RAJA.
+//
+// For details about use and distribution, please read RAJA/LICENSE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "RAJA/RAJA.hpp"
+#include "gtest/gtest.h"
+
+TEST(RangeSegmentTest, Constructors)
+{
+  RAJA::RangeSegment first(0, 10);
+  RAJA::RangeSegment copied(first);
+
+  ASSERT_EQ(first, copied);
+
+  RAJA::RangeSegment moved(std::move(first));
+
+  ASSERT_EQ(moved, copied);
+
+  RAJA::RangeSegment r1(-10, 7);
+  RAJA::RangeSegment r2(0, -50);
+}
+
+TEST(RangeSegmentTest, Assignments)
+{
+  auto r = RAJA::RangeSegment(RAJA::Index_type(), 5);
+  RAJA::RangeSegment seg1 = r;
+  ASSERT_EQ(r, seg1);
+  RAJA::RangeSegment seg2 = std::move(r);
+  ASSERT_EQ(seg2, seg1);
+}
+
+TEST(RangeSegmentTest, Swaps)
+{
+  RAJA::RangeSegment r1(0, 5);
+  RAJA::RangeSegment r2(1, 6);
+  RAJA::RangeSegment r3(r1);
+  RAJA::RangeSegment r4(r2);
+  std::swap(r1, r2);
```

Comment:
Is this just testing the copy/move ctors?

Reply:
Yes.